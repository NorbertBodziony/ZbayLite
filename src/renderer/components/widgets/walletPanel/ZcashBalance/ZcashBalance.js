import React from 'react'
import PropTypes from 'prop-types'
import BigNumber from 'bignumber.js'
import * as R from 'ramda'

import Grid from '@material-ui/core/Grid'
import Grow from '@material-ui/core/Grow'
import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

import InfoIcon from '@material-ui/icons/InfoOutlined'

import { SpinnerLoader } from '../../../ui/SpinnerLoader'
import UsdBalance from '../UsdBalance'
import ZecBalance from '../ZecBalance'

const styles = theme => ({
  title: {
    fontSize: 11
  },
  lockedTitle: {
    fontSize: '0.85rem',
    color: theme.palette.colors.white
  },
  lockedRoot: {
    height: '100%',
    backgroundColor: 'rgb(0,0,0,0.2)',
    borderRadius: 5,
    paddingTop: theme.spacing(1),
    paddingBottom: theme.spacing(1)
  },
  icon: {
    marginTop: 5,
    color: theme.typography.caption.color,
    fontSize: 25
  },
  alignHorizontal: {
    margin: 0,
    textAlign: 'center'
  },
  spinner: {
    color: theme.palette.colors.white
  },
  usd: {
    fontWeight: 700
  }
})

export const ZcashBalance = ({ classes, usdBalance, zecBalance, usdLocked, zecLocked }) => {
  return (
    <Grid container direction='row' justify='flex-start' alignItems='stretch' spacing={1}>
      <Grid item xs={12}>
        <Typography variant='body2' className={classes.title}>
          Available
        </Typography>
        <Grid container justify='space-between' align='center' direction='row'>
          <Grid item>
            <UsdBalance value={usdBalance} />
          </Grid>
          <Grid item className={classes.zec}>
            <ZecBalance size={16} value={zecBalance} style='white' />
          </Grid>
        </Grid>
      </Grid>
      {(!usdLocked.isZero() || !zecLocked.isZero()) && (
        <Grow in={!usdLocked.isZero() || !zecLocked.isZero()}>
          <Grid item xs={12}>
            <Grid
              container
              direction='row'
              justify='space-evenly'
              alignItems='center'
              className={classes.lockedRoot}
              spacing={0}
            >
              <Grid item xs={2} className={classes.alignHorizontal}>
                <SpinnerLoader classes={classes} size={30} />
              </Grid>
              <Grid item xs={7} container alignItems='center'>
                <Grid item xs={12}>
                  <Typography variant='caption' display='inline' className={classes.lockedTitle}>
                    Pending <span className={classes.usd}>${usdLocked.toFormat(2)}</span> USD
                  </Typography>
                </Grid>
                <Grid item xs={12}>
                  <Typography variant='caption'>{zecLocked.toFormat(6)} ZEC</Typography>
                </Grid>
              </Grid>
              <Grid item xs={2} className={classes.alignHorizontal}>
                <InfoIcon className={classes.icon} />
              </Grid>
            </Grid>
          </Grid>
        </Grow>
      )}
    </Grid>
  )
}
ZcashBalance.propTypes = {
  classes: PropTypes.object.isRequired,
  usdBalance: PropTypes.instanceOf(BigNumber),
  zecBalance: PropTypes.instanceOf(BigNumber),
  usdLocked: PropTypes.instanceOf(BigNumber),
  zecLocked: PropTypes.instanceOf(BigNumber)
}

ZcashBalance.defaultProps = {
  usdBalance: new BigNumber(0),
  zecBalance: new BigNumber(0),
  usdLocked: new BigNumber(0),
  zecLocked: new BigNumber(0)
}

export default R.compose(
  withStyles(styles)
)(ZcashBalance)
