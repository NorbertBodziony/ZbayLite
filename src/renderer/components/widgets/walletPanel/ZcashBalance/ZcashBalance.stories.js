import React from 'react'
// import Immutable from 'immutable'
import BigNumber from 'bignumber.js'
import { storiesOf } from '@storybook/react'
import { withKnobs, number, select, text } from '@storybook/addon-knobs'

import Paper from '@material-ui/core/Paper'

import ZcashBalance from './ZcashBalance'

storiesOf('Components/Widgets/WallePanel/ZcashBalance', module)
  .addDecorator(withKnobs)
  .add('playground', () => {
    const alignValue = select(
      'Align',
      ['flex-start', 'flex-end'],
      'flex-start'
    )
    return (
      <Paper style={{ padding: 8, width: 300 }}>
        <ZcashBalance
          title={text('title', 'Available')}
          align={alignValue}
          usdBalance={new BigNumber(number('usdBalance', 23.22))}
          zecBalance={new BigNumber(number('zecBalance', 0.2232))}
          usdLocked={new BigNumber(number('usdLocked', 1.123))}
          zecLocked={new BigNumber(number('zecLocked', 0.0045))}
        />
      </Paper>
    )
  })
