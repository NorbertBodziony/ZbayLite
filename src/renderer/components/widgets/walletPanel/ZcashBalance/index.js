import ZcashBalance from './ZcashBalance'

export default ZcashBalance
