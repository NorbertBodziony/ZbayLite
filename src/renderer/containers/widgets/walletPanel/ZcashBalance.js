import { connect } from 'react-redux'

import ZcashBalance from '../../../components/widgets/walletPanel/ZcashBalance'

import identitySelectors from '../../../store/selectors/identity'

export const mapStateToProps = state => ({
  usdBalance: identitySelectors.balance('usd')(state).plus(identitySelectors.lockedBalance('usd')(state)),
  zecBalance: identitySelectors.balance('zec')(state).plus(identitySelectors.lockedBalance('zec')(state)),
  usdLocked: identitySelectors.lockedBalance('usd')(state),
  zecLocked: identitySelectors.lockedBalance('zec')(state)
})

export default connect(mapStateToProps)(ZcashBalance)
