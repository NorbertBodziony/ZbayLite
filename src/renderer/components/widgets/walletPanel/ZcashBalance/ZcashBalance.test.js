/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'
import BigNumber from 'bignumber.js'

import { mockClasses } from '../../../../../shared/testing/mocks'
import { ZcashBalance } from './ZcashBalance'

describe('ZcashBalance', () => {
  it('renders component', () => {
    const result = shallow(
      <ZcashBalance
        classes={mockClasses}
        usdBalance={new BigNumber('3243.25')}
        zecBalance={new BigNumber('12.323212')}
      />
    )
    expect(result).toMatchSnapshot()
  })

  it('renders when balanced locked', () => {
    const result = shallow(
      <ZcashBalance
        classes={mockClasses}
        usdBalance={new BigNumber('0')}
        zecBalance={new BigNumber('0')}
        usdLocked={new BigNumber('3243.25')}
        zecLocked={new BigNumber('12.323212')}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
