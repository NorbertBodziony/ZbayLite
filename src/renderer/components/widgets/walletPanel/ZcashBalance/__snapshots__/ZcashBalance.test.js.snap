// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZcashBalance renders component 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="stretch"
  container={true}
  direction="row"
  justify="flex-start"
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Typography))
      className="title-mock"
      variant="body2"
    >
      Available
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Grid))
      align="center"
      container={true}
      direction="row"
      justify="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(UsdBalance)
          value={"3243.25"}
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="zec-mock"
        item={true}
      >
        <Memo(WithStyles(ZecBalance))
          size={16}
          style="white"
          value={"12.323212"}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`ZcashBalance renders when balanced locked 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="stretch"
  container={true}
  direction="row"
  justify="flex-start"
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Typography))
      className="title-mock"
      variant="body2"
    >
      Available
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Grid))
      align="center"
      container={true}
      direction="row"
      justify="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(UsdBalance)
          value={"0"}
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="zec-mock"
        item={true}
      >
        <Memo(WithStyles(ZecBalance))
          size={16}
          style="white"
          value={"0"}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <ForwardRef(Grow)
    in={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        className="lockedRoot-mock"
        container={true}
        direction="row"
        justify="space-evenly"
        spacing={0}
      >
        <WithStyles(ForwardRef(Grid))
          className="alignHorizontal-mock"
          item={true}
          xs={2}
        >
          <SpinnerLoader
            classes={null}
            size={30}
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          item={true}
          xs={7}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(ForwardRef(Typography))
              className="lockedTitle-mock"
              display="inline"
              variant="caption"
            >
              Pending 
              <span
                className="usd-mock"
              >
                $
                3,243.25
              </span>
               USD
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(ForwardRef(Typography))
              variant="caption"
            >
              12.323212
               ZEC
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="alignHorizontal-mock"
          item={true}
          xs={2}
        >
          <InfoOutlinedIcon
            className="icon-mock"
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </ForwardRef(Grow)>
</WithStyles(ForwardRef(Grid))>
`;
